libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -62,6 +62,18 @@ bool DeclarationTypeChecker::visit(ElementaryTypeName const& _typeName)
 	return true;
 }
 
+bool DeclarationTypeChecker::visit(EnumDefinition const& _enum)
+{
+	if (_enum.members().size() > 256)
+		m_errorReporter.fatalTypeError(
```

Comment:
Wondering whether if this error has to be fatal. Does a regular error cause problems?

Reply:
Good point. I used the same kind you get when enum has zero values but an oversized enum only affects the codegen so we could just as well let the analysis continue.

Reply:
Done. It's a non-fatal `DeclarationError` now.